src/storage/backend/sqlite/SQLite.cc

```diff
@@ -2,13 +2,58 @@
 
 #include "zeek/storage/backend/sqlite/SQLite.h"
 
+#include <thread>
+
 #include "zeek/3rdparty/sqlite3.h"
 #include "zeek/Func.h"
 #include "zeek/Val.h"
 #include "zeek/storage/ReturnCode.h"
 
+using namespace std::chrono_literals;
+
 namespace zeek::storage::backend::sqlite {
 
+OperationResult SQLite::RunPragma(std::string_view name, std::optional<std::string_view> value) {
+    int attempts = 0;
+    char* errorMsg = nullptr;
+    std::chrono::milliseconds time_spent = 0ms;
+    std::chrono::milliseconds backoff = 1ms;
+
+    std::string cmd = util::fmt("pragma %.*s", static_cast<int>(name.size()), name.data());
+    if ( value && ! value->empty() )
+        cmd += util::fmt(" = %.*s", static_cast<int>(value->size()), value->data());
+
+    while ( pragma_timeout > 0ms && time_spent < pragma_timeout ) {
+        int res = sqlite3_exec(db, cmd.c_str(), NULL, NULL, &errorMsg);
+        if ( res == SQLITE_OK ) {
+            break;
+        }
+        else if ( res == SQLITE_BUSY ) {
+            printf("busy?\n");
+            // If we got back that the database is busy, it likely means that another process is trying to
+            // do their pragmas at startup too. Exponentially back off and try again after a sleep.
+            sqlite3_free(errorMsg);
+            std::this_thread::sleep_for(backoff);
+            time_spent += backoff;
+            backoff *= 10;
```

Comment:
Wouldn't bother doing the backoff. It'll try some 500 times with the default 500ms - seems okay to me. With the backoff it's only 4 or 5 times (1, 10, 100, 1000) and all nodes retrying at similar times, potentially reducing chances to actually succeed. There's also no try after the last sleep, so might block for a long time only to return an error (less of an issue with small sleeps).

Maybe this is all theoretical anyhow, but then the exponential backoff doesn't matter either and could just retry every 1ms. Still would make the 1ms configurable.

Reply:
There's also a can of worms that if this is an async open, it would be nice to yield instead of block, but I don't think we have an "easy" primitive to do so unless I'm missing something (no need to fix).

Reply:
I added a `pragma_wait_on_failure` option and defaulted it to 5ms. 1ms even felt a little short. 500 runs of the btest though and none of them failed.